src/parser/ToInteger.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_PARSER_TOINTEGER_H
+#define SQUID_PARSER_TOINTEGER_H
+
+#include "sbuf/SBuf.h"
+#include "sbuf/Stream.h"
+
+#include <limits>
+
+namespace Parser
+{
+
+namespace Impl
+{
+
+using ParsedInteger = int64_t;
```

Comment:
Alternatively, we could extract this type from the DecimalInterger_() declaration:

```
using ParsedInteger = decltype(DecimalInteger_(std::declval<const char*>(), std::declval<const SBuf &>(),
            std::declval<int64_t>(), std::declval<int64_t>()));

```
However, it looks quite unwieldy so I'd prefer not using it.